paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/PaymentMethodService.java

```diff
@@ -0,0 +1,233 @@
+package fi.hel.verkkokauppa.payment.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import fi.hel.verkkokauppa.common.constants.OrderType;
+import fi.hel.verkkokauppa.common.constants.PaymentType;
+import fi.hel.verkkokauppa.common.error.CommonApiException;
+import fi.hel.verkkokauppa.common.error.Error;
+import fi.hel.verkkokauppa.payment.api.data.GetPaymentMethodListRequest;
+import fi.hel.verkkokauppa.payment.api.data.OrderDto;
+import fi.hel.verkkokauppa.payment.api.data.PaymentMethodDto;
+import fi.hel.verkkokauppa.payment.api.data.PaymentMethodFilter;
+import fi.hel.verkkokauppa.payment.constant.GatewayEnum;
+import fi.hel.verkkokauppa.payment.repository.PaymentMethodRepository;
+import fi.hel.verkkokauppa.payment.util.CurrencyUtil;
+import fi.hel.verkkokauppa.payment.logic.fetcher.PaymentMethodListFetcher;
+import org.helsinki.vismapay.model.paymentmethods.PaymentMethod;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@Service
+public class PaymentMethodService {
+
+	private Logger log = LoggerFactory.getLogger(PaymentMethodService.class);
+
+	private final PaymentMethodRepository paymentMethodRepository;
+	private final PaymentMethodListFetcher paymentMethodListFetcher;
+	private final Environment env;
+	private final ObjectMapper mapper;
+
+	@Autowired
```

Comment:
Changed to leverage costructor injection instead of field injection so fields can be set final.

Reply:
I like this to be like other classes in project without constructor injection. Or is there some good catch to use this? Does this cause any problems in testing?

Reply:
The reason I added the constructor injection is because we had a talk that testing would be a core part of the development process. So I thought the constructor injection would be helpful / more controlled for mocking the service instance later on in the tests. Well now that we leverage ReflectionTestUtils.setField(...) to inject field values to service instances, we do not necessarily need to have the constructor injector. 

But above that I think its a good practice to use final fields when the value is not supposed to be changed / extended. It makes the code less error prone.